taxonomy-editor-frontend/src/pages/editentry/ListTranslations.tsx

```diff
@@ -0,0 +1,183 @@
+      handleAddTranslation(languageCode);
+    });
+    setShownLanguageCodes(newShownLanguageCodes);
+    localStorage.setItem(
+      SHOWN_LANGUAGES_KEY,
+      JSON.stringify(newShownLanguageCodes)
+    );
+    setIsDialogOpen(false);
+  };
+
+  useEffect(() => {
+    // get shown languages from local storage if it exists else use main language
+    try {
+      let localStorageShownLanguages: string[] | null = localStorage.getItem(
+        SHOWN_LANGUAGES_KEY
+      )
+        ? JSON.parse(localStorage.getItem(SHOWN_LANGUAGES_KEY)!)
+        : null;
+      // validate that shown languages is an array of strings and filter all items that are valid language codes
+      if (
+        Array.isArray(localStorageShownLanguages) &&
+        localStorageShownLanguages.every((item) => typeof item === "string")
+      ) {
+        localStorageShownLanguages = localStorageShownLanguages.filter((item) =>
+          ISO6391.validate(item)
+        );
+      } else {
+        localStorageShownLanguages = [];
+      }
+
+      if (localStorageShownLanguages) {
+        // if shown languages is not empty, use it
+        localStorageShownLanguages.forEach((languageCode) => {
+          handleAddTranslation(languageCode);
+        });
+        setShownLanguageCodes(localStorageShownLanguages);
+      }
+    } catch (e) {
+      // shown languages is an empty list, when we can't parse the local storage
+      console.log(e);
+    }
+  }, [handleAddTranslation]);
+
+  const saveTranslationsForLanguage = (language: string) => {
+    return (translations: string[]) => {
+      setNodeObject((prevState) => {
+        const newNodeObject = { ...prevState };
+        newNodeObject["tags_" + language] = translations;
+        return newNodeObject;
+      });
+    };
+  };
+
+  const languagesToShow: string[] = shownLanguageCodes.filter(
+    (languageCode) => languageCode !== nodeObject.main_language
+  );
+  languagesToShow.unshift(nodeObject.main_language);
+
+  const translations: Record<string, string[]> = {};
+  Object.keys(nodeObject).forEach((key: string) => {
+    if (key.startsWith("tags") && !key.includes("ids")) {
+      const languageCode = key.split("_")[1];
+      if (languagesToShow.includes(languageCode)) {
+        translations[languageCode] = nodeObject[key].map((tag: string) => tag);
+      }
+    }
+  });
+
+  const hasFirstTranslationChanged: boolean[] = languagesToShow.map(
+    (language: string) =>
+      nodeObject[`tags_${language}`]?.[0] !==
+      originalNodeObject[`tags_${language}`]?.[0]
+  );
+
+  return (
+    <Box sx={{ ml: 4 }}>
+      {/* Title */}
+      <Stack direction="row" alignItems="center">
+        <Typography sx={{ mt: 4, mb: 1 }} variant="h5">
+          Translations
+        </Typography>
+        <Button sx={{ mt: 3.5, ml: 1 }} onClick={handleOpen}>
+          {"(" +
+            languagesToShow.length +
+            " language" +
+            (languagesToShow.length === 1 ? "" : "s") +
+            " shown)"}
```

Comment:
nit: maybe more readable to have a `numberOfLanguagesShownMessage` variable in the JSX